stdlib/public/core/Substring.swift.gyb

```diff
@@ -612,3 +612,39 @@ extension Substring {
   }
 }
 //===----------------------------------------------------------------------===//
+
+// popFirst() is only present when a collection is its own subsequence. This was
+// dropped in Swift 4.
+extension String {
+  @available(swift, obsoleted: 4, message:
```

Comment:
As discussed, it probably makes sense to add a `deprecated: 3.2` here as well. It will still compile and work in Swift 3.2 as is, but the Swift 4 compatible alternative is also available in Swift 3.2, so why not let developers know that and thus simplify the transition to Swift 4 eventually.